File: src/main/java/com/nugurang/graphql/query/EventQuery.java
package com.nugurang.graphql.query;

import com.nugurang.dto.EventDto;
import com.nugurang.service.EventService;
import graphql.kickstart.tools.GraphQLQueryResolver;
import java.util.Optional;
import org.springframework.stereotype.Service;

@Service
public class EventQuery implements GraphQLQueryResolver {
    private final EventService eventService;

    public Optional<EventDto> getEvent(Long id) {
        return Optional.ofNullable(eventService.getEvent(id)).map(entity -> entity.toDto());
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public EventQuery(final EventService eventService) {
        this.eventService = eventService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/query/EventQuery.kt
package com.nugurang.graphql.query

import com.nugurang.dto.EventDto
import com.nugurang.mapper.EventMapper
import com.nugurang.service.EventService
import graphql.kickstart.tools.GraphQLQueryResolver
import org.springframework.stereotype.Service

@Service
class EventQuery(
    private val eventService: EventService,
    private val eventMapper: EventMapper
) : GraphQLQueryResolver {

    fun getEvent(id: Long): EventDto {
        return eventService.getEvent(id).let(eventMapper::toDto)
    }
}